tests/returntypes/test_parameter_shift_shot_vec_new.py

```diff
@@ -0,0 +1,2425 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the gradients.parameter_shift module using the new return types and devices that define a shot vector."""
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.gradients import param_shift
+from pennylane.gradients.parameter_shift import _get_operation_recipe, _put_zeros_in_pdA2_involutory
+from pennylane.devices import DefaultQubit
+from pennylane.operation import Observable, AnyWires
+
+
+shot_vec_tol = 10e-3
+herm_shot_vec_tol = 0.5
+
+default_shot_vector = (1000, 2000, 3000)
+many_shots_shot_vector = tuple([1000000] * 3)
+
+# Pick 4 angles in the [-2 * np.pi, np.pi] interval
+angles = [-6.28318531, -3.92699082, 0.78539816, 3.14159265]
```

Comment:
```suggestion
angles = np.pi * (3 * np.random.random(4) - 2)
```
in case we're somehow missing something? Or even, `[-6.28318531, 3.14159265] + list(np.pi * (3 * np.random.random(2) - 2))`

Reply:
I wouldn't worry too much about a random value here. Frankly, I liked those values because they were filtered from `np.linspace`'s output used previously in tests - I think an important idea here is to make sure that things are good on the edges of the interval and somewhere at non-zero value(s) in-between.